File: python/sdk/image_antiporn_demo.py
# -*- coding:utf-8 -*-
from ais_sdk.gettoken import get_token
from ais_sdk.utils import encode_to_base64, download_url_base64
from ais_sdk.image_antiporn import image_antiporn

if __name__ == '__main__':
    user_name = '*******'
    password = '*******'
    account_name = '*******'  # the same as user_name in commonly use

    token = get_token(user_name, password, account_name)

    # call interface use the url download
    image_antiporn(token, download_url_base64('http://moderation-demo.ei.huaweicloud.com/theme/images/imagga/image_tagging_04.jpg'))

    # call interface use the url
    demo_data_url = 'https://ais-sample-data.obs.myhwclouds.com/tagging-normal.jpg'
    image_antiporn(token, '', demo_data_url )

    # call interface use the file
    image_antiporn(token, encode_to_base64('data/image-tagging-demo-1.jpg'))

File: javascript/image_antiporn_demo.js
/**
 * 图像反黄检测服务的使用示例
 */
var antiporn = require("./ais_sdk/image_antiporn");
var token = require("./ais_sdk/gettoken");
var utils = require("./ais_sdk/utils");

var username = "*******";       // 配置用户名
var domainname = "*******";     // 配置用户名
var password = "*******";       // 密码
var regionName = "cn-north-1";  // 配置区域信息

var username = "w00367249";       // 配置用户名
var domainname = "w00367249";     // 配置用户名
var password = "Ais@1234";       // 密码
var regionName = "cn-north-1";  // 配置区域信息

var filepath = "./data/moderation-antiporn.jpg";
var data = utils.changeFileToBase64(filepath);
/**
 * token 方式获取结果
 * @type {string}
 */
demo_data_url = 'https://ais-sample-data.obs.cn-north-1.myhwclouds.com/antiporn.jpg';
token.getToken(username, domainname, password, regionName, function (token) {

    antiporn.image_antiporn(token, data, "", function (result) {
        console.log(result);
    });

    antiporn.image_antiporn(token, "", demo_data_url, function (result) {
        console.log(result);
    })
});
/**
 * aksk 方式获取结果
 * @type {string}
 */
var app_key = "*************";
var app_secret = "************";

var app_key = "GVNFDUVG205CWWBQFBWG";
var app_secret = "CIKdoaSA3UQLL0iWx8ihtAZORS1akvs6ItjJyLuB";

antiporn.image_antiporn_aksk(app_key, app_secret, data, "", function (result) {
    console.log(result);
});

antiporn.image_antiporn_aksk(app_key, app_secret, "", demo_data_url, function (result) {
    console.log(result);
});